In [4]:
import os
os.chdir("/Users/srujansingh/Documents/JeanFan_Lab/Kidney_Structural_Analysis/")

In [7]:
import scanpy as sc
ctrl1_data = sc.read_h5ad("data/Ctrl_1A2_visiumhd_2um_proximal_labeled.h5ad")

AnnData object is a standardized data structure used in SC and ST datasets that hold all information at one pleace like:

Observations (obs) → cells / spots / pixels

Variables (var) → genes

Matrix (X) → expression values (genes × cells/spots)

Extra metadata → spatial coordinates, annotations, embeddings, etc.


In [8]:
ctrl1_data

AnnData object with n_obs × n_vars = 5549563 × 19059
    obs: 'in_tissue', 'array_row', 'array_col', 'pxl_row_in_fullres', 'pxl_col_in_fullres', 'proximal_tubule_geojson_feature_index'
    var: 'gene_ids', 'feature_types', 'genome'
    obsm: 'spatial'

The ctr1_data AnnData object contaisn 19059 genes (features) x 5549563 bins (spatial units).